packages/block-editor/src/components/inserter/use-async-list.js

```diff
@@ -4,9 +4,39 @@
 import { useEffect, useReducer } from '@wordpress/element';
 import { createQueue } from '@wordpress/priority-queue';
 
+/**
+ * Returns the first items from list that are present on state.
+ *
+ * @param {Array} list  New array.
+ * @param {Array} state Current state.
+ * @return {Array} First items present iin state.
+ */
+function getFirstItemsPresentInState( list, state ) {
+	const firstItems = [];
+	let index = 0;
+	let alreadyRendered = false;
+	do {
+		const item = list[ index ];
+		alreadyRendered = state.indexOf( item ) !== -1;
+		if ( alreadyRendered ) {
+			firstItems.push( item );
+		}
+		index++;
+	} while ( alreadyRendered || index === list.length );
```

Comment:
Another option could be to refactor this to break / return early if not already rendered:

```js
function getFirstItemsPresentInState( list, state ) {
	const firstItems = [];

	for ( let i = 0; i < list.length; i++ ) {
		const item = list[ i ];
		if ( ! state.includes( item ) ) {
			break;
		}

		firstItems.push( item );
	}

	return firstItems;
}
```

Reply:
Even simpler:

```js
function getFirstItemsPresentInState( list, state ) {
	const firstItems = list.filter(
		( item ) => state.includes( item )
	);

	return firstItems;
}
```

Reply:
the second proposal here doesn't give the same results. it includes all shared  items which is not the purpose of the function here.

Reply:
@youknowriad Ah, sorry. I'd confused `break` with `continue` in this instance.